### PR TITLE
DBZ-3671 Add property to exclude users in logminer query

### PR DESCRIPTION
**Pull request by infinitesnowboy**
https://issues.redhat.com/browse/DBZ-3671

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@Naros, that's one for you :)

**Review by Naros — COMMENTED** [review]
Hi @infinitesnowboy, this is actually a bit of deja-vu as I was actually contemplating this earlier today while reviewing another users' logs and thinking this might be a neat optimization :).  

Would you mind just adding a small unit test in the `LogMinerQueryBuilderTest` that uses this setting just so we have a sanity check there to make sure the SQL is built as we expect when the values are supplied?

**Comment by Naros** [comment]
@gunnarmorling @jpechane Should we consider adjusting this to support reg-ex for username matching?

**Comment by Naros** [comment]
Hi @infinitesnowboy, I went ahead and pushed 2 commits here, one including the test I requested.  Once we decide whether we want to add reg-ex support for this or not, that will then determine if this is ready to go or not.

**Comment by infinitesnowboy** [comment]
Thanks @Naros for adding the test

**Comment by Naros** [comment]
@infinitesnowboy, would you mind pulling down the few commits I pushed & rebase against master?  I'd prefer not to push a rebase to your branch if you don't mind.

**Comment by infinitesnowboy** [comment]
@Naros Sure, I rebased against master

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java [hunk 1]
```diff
@@ -304,6 +304,13 @@
             .withDescription("When set to `false`, the default, LOB fields will not be captured nor emitted. When set to `true`, the connector " +
                     "will capture LOB fields and emit changes for those fields like any other column type.");
 
+    public static final Field LOG_MINING_USERNAME_EXCLUDE_LIST = Field.create("log.mining.username.exclude.list")
+            .withDisplayName("List of users to exclude from logminer query")
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:308]:
```suggestion
            .withDisplayName("List of users to exclude from LogMiner query")
```

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java [hunk 2]
```diff
@@ -304,6 +304,13 @@
...
+            .withType(Type.STRING)
+            .withWidth(Width.SHORT)
+            .withImportance(Importance.LOW)
+            .withDescription("Comma separated list of usernames to exclude from logminer query.");
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:312]:
```suggestion
            .withDescription("Comma separated list of usernames to exclude from LogMiner query.");
```

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java [hunk 3]
```diff
@@ -401,6 +409,7 @@ public static ConfigDef configDef() {
     private final LogMiningDmlParser dmlParser;
     private final boolean archiveLogOnlyMode;
     private final boolean lobEnabled;
+    private final Set<String> logMiningUsernameExcludeList;
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:412]: Let's remove the "list" suffix; a) it's inconsistent with the actually used type, and b) it's not needed to encode the type in the name to begin with. I'd suggest `logMiningUserNameExcludes`.

**Review by gunnarmorling — COMMENTED** [review]
LGTM overall; some small suggestions inline. And the new option needs mention in _documentation/modules/ROOT/pages/connectors/oracle.adoc.

**Comment by gunnarmorling** [comment]
@Naros, if you wanted to address that one last pending comment, feel free to rebase and merge this one. I've pushed a comment for the casing change.

**Comment by gunnarmorling** [comment]
Did that last fix and applied. Thanks a lot, @infinitesnowboy!

**Pull request closed without merging**